### DISCUSSION TITLE
How can I retrieve the data stored in Azure Blob Storage to a newly installed Mimir?

### DISCUSSION THREAD

**Discussion by Ganeshpottola** [grafana/mimir#8500] [General]
Hi team,
I have installed the distributed version of Mimir (mimir-distributed-5.3.0) and configured Azure Blob Storage for it to store the data. The data is being stored every 2 hours.
My questions are:
How can I edit the time range for uploading data to Azure Blob Storage?
In case Mimir crashes, how can I retrieve the data stored in Azure Blob Storage to a newly installed Mimir?

**Comment by narqo** [comment]
**How can I edit the time range for uploading data to Azure Blob Storage?**

Since https://github.com/grafana/mimir/pull/3518 Mimir doesn't expose a way to configure TSDB block-ranges. As I can tell, after reading the discussion there, the rationale behind that went as follows:

> we're not ready for [changing TSDB block ranges period], given whenever we design features and default config we only think about the TSDB 2h block ranges period. This option was configurable only for testing purposes (e.g. we use it in integration tests) but in my opinion shouldn't be changed (at least I would love that we officially not support any value different than 2h).

**In case Mimir crashes, how can I retrieve the data stored in Azure Blob Storage to a newly installed Mimir?**

I don't fully understand this part of the question. Here how Mimir works, _per my understanding_:

Mimir's [ingester components][1] own TSDB, that accumulates incoming series and periodically cut them as "blocks" onto ingesters' local disks. Ingesters watch for newly cut blocks and upload them to an object store (e.g. Azure Blob Storage).

Whenever one queries Mimir, the [querier components][2] decide if the query requires any recent data that is still held by ingesters, or historical data, that was already propagated to the object storage (and potentially was already "massaged" by compacter component). There are numbers of "advanced" configuration knobs, that control this behaviour (e.g. [1](https://grafana.com/docs/mimir/latest/configure/configuration-parameters/#querier), [2](https://grafana.com/docs/mimir/latest/configure/configuration-parameters/#blocks_storage), [3](https://grafana.com/docs/mimir/latest/configure/configuration-parameters/#compactor)). But _typically_, changing them is not recommended unless there was an actual suggestion to do that, after someone explored the details of a particular use-case.

That is, for the question on how to make a fresh installation of Mimir to work with the existing data in an object store, you shouldn't do anything specific. As long as Mimir can access the object store, it should work. If, though, one also wants to transfer recent ("hot") blocks, which queriers expect to request from ingesters, this might be tricky. The [`mimirtool backfill`](https://grafana.com/docs/mimir/latest/manage/tools/mimirtool/#backfill) tool can backfill the data via uploading API. But I don't know if it only uploads historical data, or can also do recent one.

[1]: https://grafana.com/docs/mimir/latest/references/architecture/components/ingester/
[2]: https://grafana.com/docs/mimir/latest/references/architecture/components/querier/
